core/unit_test/CMakeLists.txt

```diff
@@ -358,6 +358,9 @@ foreach(Tag Threads;Serial;OpenMP;Cuda;HPX;OpenMPTarget;OpenACC;HIP;SYCL)
         configure_file(${dir}/dummy.cpp ${file})
         list(APPEND ${Tag}_VIEWSUPPORT ${file})
       endforeach()
+      if(Kokkos_CXX_COMPILER_ID STREQUAL "Intel")
```

Comment:
We probably should also guard this with `Kokkos_ENABLE_IMPL_MDSPAN`

Reply:
It already is a few lines up